UnitTests/TestWebSockets.cpp

```diff
@@ -167,35 +187,69 @@ void doTest(std::shared_ptr<ServerConnection> server, std::shared_ptr<DataConnec
   server.reset();
 }
 
+static bs::network::BIP15xParams getTestParams()
+{
+   bs::network::BIP15xParams params;
+   params.ephemeralPeers = true;
+   return params;
+}
+
+static bs::network::TransportBIP15xServer::TrustedClientsCallback getEmptyPeersCallback()
+{
+   return []() {
+      return bs::network::BIP15xPeers();
+   };
+}
+
+static bs::network::BIP15xPeer getPeerKey(const std::string &name, bs::network::TransportBIP15x *tr)
+{
+   return bs::network::BIP15xPeer(name, tr->getOwnPubKey());
+}
+
+static bs::network::BIP15xPeer getPeerKey(const std::string &host, const std::string &port
+   , bs::network::TransportBIP15x *tr)
+{
+   std::string name = fmt::format("{}:{}", host, port);
+   return bs::network::BIP15xPeer(name, tr->getOwnPubKey());
+}
+
+
 TEST_F(TestWebSocket, Basic)
 {
+   const auto &srvTransport = std::make_shared<bs::network::TransportBIP15xServer>(
+      StaticLogger::loggerPtr, getEmptyPeersCallback());
+   auto server = std::make_shared<WsServerConnection>(StaticLogger::loggerPtr, srvTransport);
 
-   WsServerConnectionParams serverParams;
-   auto server = std::make_shared<WsServerConnection>(StaticLogger::loggerPtr, serverParams);
+   const auto &clientTransport = std::make_shared<bs::network::TransportBIP15xClient>(
+      StaticLogger::loggerPtr, getTestParams());
+   auto client = std::make_shared<WsDataConnection>(StaticLogger::loggerPtr, clientTransport);
 
-   WsDataConnectionParams clientParams;
-   clientParams.useSsl = false;
-   auto client = std::make_shared<WsDataConnection>(StaticLogger::loggerPtr, clientParams);
+   srvTransport->addAuthPeer(getPeerKey("client", clientTransport.get()));
+   clientTransport->addAuthPeer(getPeerKey(kTestTcpHost, kTestTcpPort, srvTransport.get()));
 
    doTest(std::move(server), std::move(client), kTestTcpHost, kTestTcpPort, kTestTcpHost, kTestTcpPort);
 }
 
 TEST_F(TestWebSocket, Router)
 {
-   WsServerConnectionParams serverParams;
+   const auto &srvTransport = std::make_shared<bs::network::TransportBIP15xServer>(
+      StaticLogger::loggerPtr, getEmptyPeersCallback());
 
    RouterServerConnectionParams::Server server1;
    server1.host = kTestTcpHost;
    server1.port = kTestTcpPort;
-   server1.server = std::make_shared<WsServerConnection>(StaticLogger::loggerPtr, serverParams);
+   server1.server = std::make_shared<WsServerConnection>(StaticLogger::loggerPtr, srvTransport);
 
    RouterServerConnectionParams routerServerParams;
    routerServerParams.servers.push_back(std::move(server1));
    auto server = std::make_shared<RouterServerConnection>(StaticLogger::loggerPtr, routerServerParams);
 
-   WsDataConnectionParams clientParams;
-   clientParams.useSsl = false;
```

Comment:
why SSL is disabled? staging proxy works behind SSL-ed nginx, and so this will break it

Reply:
Proxy will work via BIP15x and WebSockets, this was the main objective of this (and related) PR.